include/pybind11/stl/filesystem.h

```diff
@@ -13,21 +13,30 @@
 #include <string>
 
 #ifdef __has_include
-#    if defined(PYBIND11_CPP17) && __has_include(<filesystem>)
-#        include <filesystem>
-#        define PYBIND11_HAS_FILESYSTEM 1
+#    if defined(PYBIND11_CPP17)
+#        if __has_include(<filesystem>) && \
+          PY_VERSION_HEX >= 0x03060000
+#            include <filesystem>
+#            define PYBIND11_HAS_FILESYSTEM 1
+namespace fs = std::filesystem;
```

Comment:
This is in the global namespace, and a very short name => I'm worried about name clashes. Could you please make this longer, e.g. `pybind11_filesystem_alias` or similar?

Reply:
Hey, great catch. I'll go ahead and update it.